org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/CompletionProposalRequestor.java

```diff
@@ -430,6 +430,19 @@ public CompletionItem toCompletionItem(CompletionProposal proposal, int index) {
 			} else if (labelDetailsEnabled && $.getKind() == CompletionItemKind.Constructor && $.getLabelDetails() != null && $.getLabelDetails().getDetail() != null) {
 				filterText = newText.concat($.getLabelDetails().getDetail());
 			}
+			if (proposal.getKind() == CompletionProposal.METHOD_REF
+					&& collapsedTypes.containsKey(String.valueOf(proposal.getName())) && collapsedTypes.get(String.valueOf(proposal.getName())) > 1) {
```

Comment:
Could you use `collapsedTypes.getOrDefault(String.valueOf(proposal.getName()), 0) > 1` to avoid the first `containsKey` check ?